pandas/io/json/_json.py

```diff
@@ -982,7 +982,7 @@ def _try_convert_to_date(self, data):
         for date_unit in date_units:
             try:
                 new_data = to_datetime(new_data, errors="raise", unit=date_unit)
-            except (ValueError, OverflowError):
+            except (ValueError, OverflowError, TypeError):
```

Comment:
@jbrockmendel any thoughts here?

Reply:
the usual: if we can handle this case before the try/except that would be preferable

Reply:
@jbrockmendel I’ll give that approach a try.

Reply:
@jbrockmendel Pushed an alternate approach that escapes from the _try_convert_to_date() method if the dtype is bool, avoiding the try/except but adding a conditional to the method (which I don't love, but at least the method already has dtype-based conditionals).

Reply:
yes this is ok. an alternative approach is to `infer_dtype` on this and then dispatch. If someone wants to try that (later PR) would be ok.